<a href="https://colab.research.google.com/github/macs1903/Devoir_INF8100/blob/main/INF8100_TP1_partie2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd
import numpy as np

salaries = pd.read_csv("https://raw.githubusercontent.com/macs1903/Devoir_INF8100/main/salaires.csv", encoding='latin-1')

In [3]:
salaries.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32561 entries, 0 to 32560
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   age                 32561 non-null  int64 
 1   classe.travail      32561 non-null  object
 2   education           32561 non-null  object
 3   nombre.education    32561 non-null  int64 
 4   etat.civil          32561 non-null  object
 5   occupation          32561 non-null  object
 6   Lien.parente        32561 non-null  object
 7   race                32561 non-null  object
 8   sexe                32561 non-null  object
 9   heures.par.semaine  32561 non-null  int64 
 10  pays.natal          32561 non-null  object
 11  salaire             32561 non-null  object
dtypes: int64(3), object(9)
memory usage: 3.0+ MB


In [3]:
len(salaries)

32561

#***Ici nous allons renommer les colonnes de notre dataset.

In [4]:
salaries.rename(columns={'classe.travail': 'classe_travail',
                         'nombre.education': 'nombre_education',
                         'etat.civil': 'etat_civil',
                         'Lien.parente': 'lien_parente',
                         'heures.par.semaine': 'heures_par_semaine',
                         'pays.natal': 'pays_natal'},
                inplace=True)

# 2. Combien de valeurs manquantes y a-t-il pour chaque colonne ?

La plupart de temps les valeurs manquantes sont representes par **NAs**, mais d'autrefois elles peuvent etre representés par d'autres caractères specifiques comme **?** ou une valeur beaucoup plus grande

In [5]:
salaries.replace('?', np.nan, inplace=True)

In [6]:
salaries.isna().sum()

age                      0
classe_travail        1836
education                0
nombre_education         0
etat_civil               0
occupation            1843
lien_parente             0
race                     0
sexe                     0
heures_par_semaine       0
pays_natal             583
salaire                  0
dtype: int64

Les colonnes contenant des valeurs manquante sont:
* classe.travail        1836
* occupation            1843
* pays.natal             583



Quelle est la colonne (feature) qui a le plus de valeurs manquantes ?

la colonne de **1843**

#3. Combien d’hommes y a-t-il dans ce jeu de données ?

In [7]:
#salaries.loc[salaries['sexe']=='Masculin',].value_counts()
nbHomme = salaries['sexe'].value_counts()['Masculin']
print("Le nombre total d'homme est de : ", nbHomme)

Le nombre total d'homme est de :  21790


# 4. Quel est l’âge le plus élevé et le plus bas de ce jeu de données ?

In [8]:
ageMin = salaries['age'].min()
ageMax = salaries['age'].max()

print("L'âge le plus bas est : ", ageMin)
print("L'âge le plus élévé est : ", ageMax)

L'âge le plus bas est :  17
L'âge le plus élévé est :  90


# 5. Combien de femmes travaillent dans le secteur privé?

In [9]:
femmeSecPrive = salaries.loc[(salaries['sexe']=='Feminin') & (salaries['classe_travail']=='Privee'), ]
salaries.groupby('classe_travail')['sexe'].value_counts()['Privee']

sexe
Masculin    14944
Feminin      7752
Name: sexe, dtype: int64

# 6. Quelle est la liste des différents pays représentés dans la colonne représentant le pays natal/d’origine de chaque personne de notre jeu de données?

In [11]:
paysNatal = salaries['pays_natal'].dropna()
list(paysNatal.unique())

['United-States',
 'Mexico',
 'Greece',
 'Vietnam',
 'China',
 'Taiwan',
 'India',
 'Philippines',
 'Trinadad&Tobago',
 'Canada',
 'South',
 'Holand-Netherlands',
 'Puerto-Rico',
 'Poland',
 'Iran',
 'England',
 'Germany',
 'Italy',
 'Japan',
 'Hong',
 'Honduras',
 'Cuba',
 'Ireland',
 'Cambodia',
 'Peru',
 'Nicaragua',
 'Dominican-Republic',
 'Haiti',
 'El-Salvador',
 'Hungary',
 'Columbia',
 'Guatemala',
 'Jamaica',
 'Ecuador',
 'France',
 'Yugoslavia',
 'Scotland',
 'Portugal',
 'Laos',
 'Thailand',
 'Outlying-US(Guam-USVI-etc)']

# 6.3 Algèbre linéaire (6pts)
Pour les questions de cette partie, nous allons uniquement travailler avec une petite partie des données : les femmes du jeu de données. Nous allons garder le 1/10 des données pour les tests.


#6.4 Analyses statistiques et probabilistes (12pts)

0. On va convertir le revenu(salaire) en entier.

1. Quelle est la moyenne des âges ? Quelle est la moyenne du nombre d’heures travaillées par semaine ? et les valeurs médianes ?

In [12]:
salaries['salaire'].replace('<=50K', 40000, inplace=True)
salaries['salaire'].replace('>50K', 60000, inplace=True)

In [13]:
# 1. Moyennes et médianes des âges et des heures de travail
mean_age = salaries['age'].mean()
median_age = salaries['age'].median()
mean_hours_per_week = salaries['heures_par_semaine'].mean()
median_hours_per_week = salaries['heures_par_semaine'].median()
print("1. Moyenne des âges :", mean_age)
print("   Médiane des âges :", median_age)
print("   Moyenne des heures travaillées par semaine :", mean_hours_per_week)
print("   Médiane des heures travaillées par semaine :", median_hours_per_week)

# 2. Âges les plus représentatifs
most_representative_ages = salaries['age'].value_counts().head(5)
print("\n2. Âges les plus représentatifs :")
print(most_representative_ages)

# 3. Variance des heures de travail et des âges
variance_hours_per_week = salaries['heures_par_semaine'].var()
variance_age = salaries['age'].var()
print("\n3. Variance des heures travaillées par semaine :", variance_hours_per_week)
print("   Variance des âges :", variance_age)

# 4. Impact de l'éducation vs. âge sur le salaire
education_vs_age_salary = salaries.groupby(['education', 'age'])['salaire'].mean()
print("\n4. Impact de l'éducation vs. âge sur le salaire :")
print(education_vs_age_salary)

# 5. Impact de l'éducation vs. heures de travail sur le salaire
education_vs_hours_per_week_salary = salaries.groupby(['education', 'heures_par_semaine'])['salaire'].mean()
print("\n5. Impact de l'éducation vs. heures de travail sur le salaire :")
print(education_vs_hours_per_week_salary)

# 6. Impact de l'âge sur le salaire
average_age_below_50k = salaries[salaries['salaire'] == '<= 50K']['age'].mean()
average_age_above_50k = salaries[salaries['salaire'] == '> 50K']['age'].mean()
print("\n6. Impact de l'âge sur le salaire :")
print("   Âge moyen pour <= 50K :", average_age_below_50k)
print("   Âge moyen pour > 50K :", average_age_above_50k)

# 7. Impact des heures de travail sur le salaire
average_hours_below_50k = salaries[salaries['salaire'] == '<= 50K']['heures_par_semaine'].mean()
average_hours_above_50k = salaries[salaries['salaire'] == '> 50K']['heures_par_semaine'].mean()
print("\n7. Impact des heures de travail sur le salaire :")
print("   Heures moyennes pour <= 50K :", average_hours_below_50k)
print("   Heures moyennes pour > 50K :", average_hours_above_50k)

# 8. Différence de salaire entre personnes mariées et célibataires
marital_status_salary = salaries.groupby(['etat_civil'])['salaire'].mean()
print("\n8. Différence de salaire entre personnes mariées et célibataires :")
print(marital_status_salary)

# 9. Proportion des hommes et des femmes ayant un doctorat et gagnant plus de 50k
proportion_doctorate_men = salaries[(salaries['sexe'] == 'Masculin') & (salaries['education'] == 'Doctorat') & (salaries['salaire'] == '> 50K')].shape[0] / num_men
proportion_doctorate_women = salaries[(salaries['sexe'] == 'Feminin') & (salaries['education'] == 'Doctorat') & (salaries['salaire'] == '> 50K')].shape[0] / (salaries['sexe'] == 'Feminin').sum()
print("\n9. Proportion des hommes avec un doctorat gagnant > 50K :", proportion_doctorate_men)
print("   Proportion des femmes avec un doctorat gagnant > 50K :", proportion_doctorate_women)

# 10. Proportion des femmes mariées sans master ni doctorat
#proportion_married_women_without_masters_or_doctorate = data[(data['sexe'] == 'feminin') & (data['etat_civil'] == 'Married-civ-spouse') & (data['Education'] != 'Masters') & (data['Education'] != 'Doctorate')].shape[0] / (data[(data['Sexe'] == 'Female') & (data['Etat civil'] == 'Married-civ-spouse')].shape[0]
#print("\n10. Proportion des femmes mariées sans master ni doctorat :", proportion_married_women_without_masters_or_doctorate)

# 11. Pays avec les salaires les plus élevés
#highest_earning_country = data[data['Revenu'] == '> 50K']['Pays d'origine'].value_counts().idxmax()
#print("\n11. Pays avec les salaires les plus élevés :", highest_earning_country)


1. Moyenne des âges : 38.58164675532078
   Médiane des âges : 37.0
   Moyenne des heures travaillées par semaine : 40.437455852092995
   Médiane des heures travaillées par semaine : 40.0

2. Âges les plus représentatifs :
36    898
31    888
34    886
23    877
35    876
Name: age, dtype: int64

3. Variance des heures travaillées par semaine : 152.45899505045415
   Variance des âges : 186.06140024880162

4. Impact de l'éducation vs. âge sur le salaire :
education     age
10th          17     40000.0
              18     40000.0
              19     40000.0
              20     40000.0
              21     40000.0
                      ...   
Some-college  81     40000.0
              82     40000.0
              84     60000.0
              88     40000.0
              90     40000.0
Name: salaire, Length: 965, dtype: float64

5. Impact de l'éducation vs. heures de travail sur le salaire :
education     heures_par_semaine
10th          1                     40000.000000
              2

NameError: ignored